### PR TITLE
Support reading events from Kafka

### PR DESCRIPTION
**Pull request by hmadison**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area plugins

**What this PR does / why we need it**:

This makes an attempt to resolve a line item in #228 around Kafka support. The changes adds a new plugin which allows for a consumer group to read the “bytes off the wire” from a given set of Kafka topics. It explicitly does not support parsing these bytes since Kafka’s internal model is inherently byte based. Instead, we can delegate the parsing to extractor plugins which can build on top of the events provided by this plugin.

**Which issue(s) this PR fixes**:

In part #228.

**Special notes for your reviewer**:
* From a test perspective, this makes use of testcontainers/docker to validate the behavior of the plugin. I’m not sure if this work under the current test infrastructure for plugins.
* This seems to have some event time sensitivity issues. At low levels of traffic messages are being read in the go routine but are dropped downstream.

### CODE REVIEW

**Comment by leogr** [comment]
Hey @hmadison 

Welcome and thank you for this PR! :) 

Just a couple of questions:
- Could you like to [register](https://github.com/falcosecurity/plugins?tab=readme-ov-file#registering-a-new-plugin) your plugin and pick the next available ID?
- Are you willing to maintain this plugin in the long-term? If yes, I'd recommend proposing yourself as [reviewer](https://github.com/falcosecurity/evolution/blob/main/MAINTAINERS_GUIDELINES.md#onboarding-a-reviewer) for this plugin.

cc @falcosecurity/plugins-maintainers

Also, I'm not a Kafka expert, so I'm not likely the right person to review this PR. Please, be patient we will find someone :) 

Thank you :pray:

**Comment by hmadison** [comment]
Hi @leogr,

I'm fine maintaining this long term and helping with review.

Thanks,
Hunter

**Comment by leogr** [comment]
> Hi @leogr,
> 
> I'm fine maintaining this long term and helping with review.
> 
> Thanks, Hunter

This is great! So let's proceed this way:
- We will review and merge this PR (it will take a bit, but we do our best). Cc @Issif @alacuku. May you help with this?
- Once the PR is merged, you will add yourself as a reviewer in a separate PR (and I will sponsor you), then we will do the [onboarding](https://github.com/falcosecurity/evolution/blob/main/MAINTAINERS_GUIDELINES.md#onboarding-a-reviewer) 

Would this work for you? :)

File: registry.yaml
Comment (leogr) [inline registry.yaml:461]:
```suggestion
      - name: The Falco Authors
        email: cncf-falco-dev@lists.cncf.io
      - name: Hunter Madison
        email: Hunter.Madison@ibm.com
```

**Comment by Issif** [comment]
Hi,

I'll take a look when free, it can take a week or 2.

**Comment by hmadison** [comment]
@leogr,

Works great!

Thanks,
Hunter

File: OWNERS
Comment (Issif) [inline OWNERS:1]: Don't modify this file please.

**Review by Issif — COMMENTED** [review]
Can you write a README to describe the configuration, how to install/use this plugin, please. An OWNERS files too, with your nickname to allow you to review the future PR.

**Comment by leogr** [comment]
Also, this test is constantly failing, can you take a look?
Screenshot from 2024-04-30 18-49-10

cc @alacuku @Andreagit97 @FedeDP. The failure might not be related to this PR. Could you please take a look at it, too? :pray:

File: plugins/kafka/OWNERS
Comment (leogr) [inline plugins/kafka/OWNERS:2]: Hey @hmadison 

With this your are proposing yourself as a [maintainer](https://github.com/falcosecurity/evolution/blob/main/MAINTAINERS_GUIDELINES.md#onboarding-a-maintainer), not just a [reviewer](https://github.com/falcosecurity/evolution/blob/main/MAINTAINERS_GUIDELINES.md#onboarding-a-reviewer). 

Either option is acceptable, but onboarding processes and responsibilities are different. 
I'd like to check with you about the differences. and you are selecting the wanted option:

- (a) If you are proposing yourself to be a maintainer, your current change is okay. In such a case, please comment below and confirm this is what you want. I will take care of the onboarding process.
- or (b), if you are proposing yourself as a reviewer only, change `approvers` :point_right: `reviewers.`


/hold

Comment (hmadison) [reply plugins/kafka/OWNERS]: Sounds good! I'm after option "b". Sorry for the confusion!

**Comment by alacuku** [comment]
> Also, this test is constantly failing, can you take a look? Screenshot from 2024-04-30 18-49-10
> 
> cc @alacuku @Andreagit97 @FedeDP. The failure might not be related to this PR. Could you please take a look at it, too? 🙏

It seems related to the go version:
`go: go.mod requires go >= 1.22.2 (running go 1.21.9; GOTOOLCHAIN=local)`

@hmadison, could you please bump the go version used in our CI?

**Comment by leogr** [comment]
Hey @hmadison 

Unfortunately, we do not allow [merge commits](https://github.com/falcosecurity/plugins/pull/467/commits/f1eb4036af26b19cc921a00a266605590ecc7145) in our git history.

Could you [rebase](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md#resolving-conflicts-by-rebasing) this PR and remove the merge commit, pls? :pray:

**Review by leogr — APPROVED** [review]
Approved!

:partying_face: 

Thank you!

**Comment by leogr** [comment]
Also:

I'm going to release the plugin next week

**Pull request merged**
